aspnetcore/security/blazor/server-side.md

```diff
@@ -0,0 +1,378 @@
+  * At the app level:
+    * Endpoint routing extensibility.
+    * Require authentication to connect to the app and keep track of the active sessions per user.
+    * Reject new sessions upon reaching a limit.
+    * Proxy WebSocket connections to an app through the use of a proxy, such as the [Azure SignalR Service](/azure/azure-signalr/signalr-overview) that multiplexes connections from clients to an app. This provides an app with greater connection capacity than a single client can establish, preventing a client from exhausting the connections to the server.
+  * At the server level: Use a proxy/gateway in front of the app. For example, [Azure Front Door](/azure/frontdoor/front-door-overview) enables you to define, manage, and monitor the global routing of web traffic to an app.
+
+<!-- Endpoint routing extensibility: Do we doc this somewhere that we can cross-link? -->
+
+## Denial of service (DoS) attacks
+
+<!-- I moved this here because DoS seems to flow after server resource exhaustion. I understand that not ALL DoS attacks exhaust resources, so I made this an H2 here (i.e., it's not directly under Resource Exhaustion). -->
+
+Denial of service (DoS) attacks involve a client causing the server to exhaust one or more of its resources making the app unavailable. Blazor server-side apps include some default limits and rely on other ASP.NET Core and SignalR limits to protect against DoS attacks:
+
+| Blazor server-side app limit                            | Description | Default |
+| ------------------------------------------------------- | ----------- | ------- |
+| `CircuitOptions.DisconnectedCircuitMaxRetained`         | Maximum number of disconnected circuits that a given server holds in memory at a time. | 100 |
+| `CircuitOptions.DisconnectedCircuitRetentionPeriod`     | Maximum amount of time a disconnected circuit is held in memory before being torn down. | 3 minutes |
+| `CircuitOptions.JSInteropDefaultCallTimeout`            | Maximum amount of time the server waits before timing out an asynchronous JavaScript function invocation. | 1 minute |
+| `CircuitOptions.MaxBufferedUnacknowledgedRenderBatches` | Maximum number of unacknowledged render batches the server keeps in memory per circuit at a given time to support robust reconnection. After reaching the limit, the server stops producing new render batches until one or more batches have been acknowledged by the client. | 10 |
+
+
+| SignalR and ASP.NET Core limit             | Description | Default |
+| ------------------------------------------ | ----------- | ------- |
+| `CircuitOptions.MaximumReceiveMessageSize` | Message size for an individual message. | 32 KB |
+
+<!-- The original text only lists one SignalR limit (MaximumReceiveMessageSize). Is this the only one that you'd like to show here. Should we eliminate this little one-line table and just cross-link to https://docs.microsoft.com/aspnet/core/signalr/configuration#configure-server-options? -->
```

Comment:
I think that this limit in particular is pretty important, I don't want to see it go away.